hal/src/nRF52840/sleep_hal.cpp

```diff
@@ -973,6 +966,17 @@ static int enterStopBasedSleep(const hal_sleep_config_t* config, hal_wakeup_sour
     __DSB();
     __ISB();
 
+    // WARNING: This is going to fix the issue that when both .ble() and .duration()
+    // are specified as wakeup sources, device will run into hardfault if the following
+    // check comes after __disable_irq(), such as in configRtcWakeupSource() previously.
+    nrf_rtc_event_enable(NRF_RTC2, RTC_EVTEN_TICK_Msk);
+    // Make sure that RTC is ticking
+    // See 'TASK and EVENT jitter/delay'
+    // http://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.nrf52840.ps%2Frtc.html
+    while (!nrf_rtc_event_pending(NRF_RTC2, NRF_RTC_EVENT_TICK));
```

Comment:
I guess this one takes a long time to complete?

Reply:
yup. I replaced the `while()` loop with `HAL_Delay_Microseconds(1000000)` and it crashed as well. If I shorten the delay, it worked as expected. So it's weird that why the `NRF_RTC_EVENT_TICK` event is not asserted after `__disable_irq()` when softdevice is enabled.